w3af/plugins/audit/hpp.py

```diff
@@ -0,0 +1,263 @@
+                precedence = FIRST
+                self.client_side(freq, orig_response, precedence, value, param)
+                continue
+
+            if fuzzy_equal(rtest_resp.body, ptest_body, EQUALITY_THRESHOLD):
+                precedence = LAST
+                self.client_side(freq, orig_response, precedence, value, param)
+                continue
+
+            concat = find_reflection(ptest_resp, None, [value, newval])
+
+            if concat:
+                desc = u'Concatenation of values of the two input parameters ' \
+                       u'with the same name \"%s\" detected inside HTML ' \
+                       u'code of response body.\nThe request url was %s.\n' \
+                       u'Got concatenated output reflected in this link %s\n' \
+                       u'This \"feature\" may be used in different WAF ' \
+                       u'bypassing techniques.'
+                desc = desc % (param, ptest_url, concat[1])
+
+                i = Info(u'Input concatenation detected', desc, ptest_resp.id,
+                         self.get_name())
+                i.set_url(ptest_url)
+                i.precedence = CONCAT
+                self.kb_append(self, 'hpp', i)
+                return
+
+            om.out.debug(u'Unable to determine precedence for HPP detection '
+                         u'with this request %s. Presumably not vulnerable to'
+                         u' HPP.' % ptest_url)
+            return
+
+    def client_side(self, freq, orig_response, precedence, value, param):
+
+        reflections = find_reflection(orig_response, precedence, [value])
+
+        for reflection in reflections:
+            #getting the list of exploitable params
+            forgeable = get_forgeable(reflection, precedence)
+
+            for f in forgeable:
+                init_val = reflection[1].get_querystring()[f][0]
+                if init_val.isdigit():
+                    forged_val = str(int(init_val)+1)
+                else:
+                    forged_val = FORGE_TOKEN
+                #building attack payload and exploiting
+                payloads = [unicode(value) + '&' + f + '=' + forged_val]
+
+                mutants = create_mutants(freq, payloads,
+                                         fuzzable_param_list=param,
+                                         mutant_tuple=(QSMutant,))
+                mu = mutants[0]
+
+                vtest_resp = self._uri_opener.send_mutant(mu)
+                #checkig for successful exploitation
+                r = find_reflection(vtest_resp, precedence, [forged_val])
+                if r:
+                    msg = u"The %s uri is vulnerable to HPP attack: value of " \
+                          u"parameter %s can be forged using URL: %s, " \
+                          u"precedence is %s"
+                    msg = msg % (freq.get_uri(), r, mu.get_uri(), precedence)
+                    v = Vuln.from_mutant(u'HPP', desc=msg, mutant=mu,
```

Comment:
Use full name for vuln, the user might not know what "HPP" is and can't google that. Use "HTTP Parameter Pollution" instead.